### PR TITLE
Added chars counter directive

### PR DESCRIPTION
**Pull request by Tilesto**

### CODE REVIEW

File: VirtoCommerce.Platform.Web/Scripts/common/directives/chars-count.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿angular.module('platformWebApp')
+    .directive('vaCharsCount', function () {
+        return {
+            template: `<div class="chars-counter">
+                           <span>{{vaField == undefined ? '0' : vaField.length}}</span> / <span>{{vaMaxlength}}</span>
```
Comment (tatarincev) [inline VirtoCommerce.Platform.Web/Scripts/common/directives/chars-count.js]: `vaField == undefined` can be replaced to `vaField ? ` or `if (vaField)` or better `!!vaField`

File: VirtoCommerce.Platform.Web/Scripts/common/directives/chars-count.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿angular.module('platformWebApp')
+    .directive('vaCharsCount', function () {
+        return {
+            template: `<div class="chars-counter">
+                           <span>{{vaField == undefined ? '0' : vaField.length}}</span> / <span>{{vaMaxlength}}</span> 
+                       </div>`,
+            scope: {
+                vaField: "=",
```
Comment (tatarincev) [inline VirtoCommerce.Platform.Web/Scripts/common/directives/chars-count.js]: Need to use one-way binding '@'

File: VirtoCommerce.Platform.Web/Scripts/common/directives/chars-count.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+                vaMaxlength: "@"
+            },
+            link: function (scope, element) {
+                scope.$parent.vaMaxlength = scope.vaMaxlength;
```
Comment (tatarincev) [inline VirtoCommerce.Platform.Web/Scripts/common/directives/chars-count.js]: I think $parent there is redundant, why it is necessary here?

File: VirtoCommerce.Platform.Web/Content/themes/main/css/main.css [hunk 4]
```diff
@@ -86,10 +86,21 @@ fieldset {
...
     display: table-cell; } }
 
+/*Chars counter label for field*/
+.chars-counter{
+    display: inline-block;
+    float: right;
+    font-size: 14px;
+}
+
+.invalid{
```
Comment (n2pro) [inline VirtoCommerce.Platform.Web/Content/themes/main/css/main.css]: add appropriate colors for states: info, warning and other

File: VirtoCommerce.Platform.Web/Scripts/common/directives/chars-count.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+﻿angular.module('platformWebApp')
+    .directive('vaCharsCount', function () {
+        return {
+            template: `<div class="chars-counter">
```
Comment (n2pro) [inline VirtoCommerce.Platform.Web/Scripts/common/directives/chars-count.js]: use $templateCache

File: VirtoCommerce.Platform.Web/Scripts/common/directives/chars-count.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+                            <span>{{!!vaField ? vaField.length : 0}}</span> / <span>{{vaMaxlength}}</span> 
+                       </div>`,
+            scope: {
+                vaField: "@",
+                vaMaxlength: "@"
+            },
+            link: function (scope, element) {
+                scope.$watch('vaField', function () {
+                    if (scope.vaField.length > scope.vaMaxlength) {
+                        element.addClass('invalid');
```
Comment (n2pro) [inline VirtoCommerce.Platform.Web/Scripts/common/directives/chars-count.js]: can be used toggleClass() depending on the va-model

**Pull request merged**